libs/iostability/iostability_lib.py

```diff
@@ -124,19 +128,53 @@ def __init__(self, sender, receiver, test_id, health_obj):
         self.event_pass = threading.Event()
         self.event_fail = threading.Event()
         self.mail_obj = Mail(sender=sender, receiver=receiver)
+        self.sender = sender
+        self.receiver = receiver
         self.test_id = test_id
         self.health_obj = health_obj
         self.interval = 4  # Mail to be sent periodically after 4 hours.
+        self.message_id = None
+        self.build_url = build_url
+        self.start_time = datetime.now()
+
+    def prepare_email(self, execution_status) -> MIMEMultipart:
+        """
+        Prepare email message with format and attachment
+        :param execution_status: Execution status. In Progress/Fail
+        :return: Formatted MIME message
+        """
+        hctl_status = json.dumps(self.health_obj.hctl_status_json(), indent=4)
+        status = f"IOStability: {self.test_id} {execution_status} on {self.health_obj.hostname}"
+        subject = status
+        body = f"<p>Hours of execution: {datetime.now() - self.start_time}\n<p>"
+        if self.build_url:
+            body += f"""Jenkins Job: <a href="{self.build_url}">{self.build_url}</a>\n"""
```

Comment:
We can get jenkins_url from environment here, itself, instead of passing it through many function calls/parameters.

Reply:
Done.